src/entities/DealTokenSwap.ts

```diff
@@ -140,6 +140,13 @@ export class DealTokenSwap implements IDeal {
   //   return;
   // }
 
+  get timeLeftToExecute(): number | undefined {
+    if (!this.executedAt) {
+      return;
+    }
+    return this.executedAt.getTime() + this.executionPeriod * 1000 - new Date().getTime();
```

Comment:
Specify parentheses.

Use `Date.now()`; is cleaner, probably more efficient.

add @computedFrom